src/components/budget/budget.service.ts

```diff
@@ -419,19 +419,33 @@ export class BudgetService {
   ): Promise<Budget> {
     const budget = await this.readOne(input.id, session);
 
-    await this.files.updateDefinedFile(
-      budget.universalTemplateFile,
-      'budget.universalTemplateFile',
-      universalTemplateFile,
-      session
+    const realChanges = await this.db.getActualChanges(
+      budget,
+      input,
+      Budget.Props
     );
-
-    return await this.db.sgUpdateProperties({
-      session,
+    await this.authorizationService.verifyCanEditChanges(
+      Budget,
+      budget,
+      realChanges
+    );
+    if (universalTemplateFile) {
+      await this.authorizationService.verifyCanEdit({
+        resource: Budget,
+        baseNode: budget,
+        prop: 'universalTemplateFile',
+      });
```

Comment:
Or maybe since defined files are a common thing the `getActualChanges` can account for that and we don't spread them out until auth verification.

I know there's the mismatch in name with relation modifications. Those would still be necessary like you have them.